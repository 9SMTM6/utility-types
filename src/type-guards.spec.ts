import { Primitive, Falsey } from './mapped-types';
import { isPrimitive, isFalsey } from './type-guards';

// @dts-jest:group isPrimitive
it('narrows to correct type', () => {
  const consumer = (param: Primitive[] | Primitive): string => {
    if (isPrimitive(param)) {
      // @dts-jest:pass:snap
      param;
      return String(param) + ' was Primitive';
    }
    // @dts-jest:pass:snap
    param;
    const resultArray = param
      .map(consumer)
      .map(rootString => '\n\t' + rootString);
    return resultArray.reduce((comm, newV) => comm + newV, 'this was nested:');
  };
});

// @dts-jest:group isFalsey-Group-Falsey
it('returns true for falsey and narrows type', () => {
  const falseyTestVals: unknown[] = ['', null, undefined, false, 0];

  falseyTestVals.forEach(val => {
    if (isFalsey(val)) {
      // @dts-jest:pass:snap
      val;
    }
    // @dts-jest:pass:snap
    val;
  });

  const testResults = falseyTestVals.map(isFalsey);

  testResults.forEach(val => expect(val).toBe(true));
});

// @dts-jest:group isFalsey-Group-Truthy
it('returns false for truthy and narrows type', () => {
  const truthyTestVals: unknown[] = [' ', true, {}, []];

  truthyTestVals.forEach(val => {
    if (isFalsey(val)) {
      // @dts-jest:pass:snap
      val;
    }
    // @dts-jest:pass:snap
    val;
  });

  const testResults = truthyTestVals.map(isFalsey);

  testResults.forEach(val => expect(val).toBe(false));
});
