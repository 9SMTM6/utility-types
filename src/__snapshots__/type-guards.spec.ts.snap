// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`isFalsey-Group-Falsey val 1`] = `"Falsey"`;

exports[`isFalsey-Group-Falsey val 2`] = `"unknown"`;

exports[`isFalsey-Group-Truthy val 1`] = `"Falsey"`;

exports[`isFalsey-Group-Truthy val 2`] = `"unknown"`;

exports[`isPrimitive param 1`] = `"Primitive"`;

exports[`isPrimitive param 2`] = `"Primitive[]"`;
